mux/tree.go

```diff
@@ -71,22 +72,22 @@ func (t Tree) Compile() Tree {
 // Match path to Node
 func (t Tree) Match(path string) (Node, middleware.Middleware, context.Params, string) {
 	for _, child := range t {
-		if node, m, params, subPath := child.Match(path); node != nil {
+		if node, m, params, subPath := child.Match(path); node != nil && node.Route() != nil {
```

Comment:
` && node.Route() != nil` we should not check here the value of route, because if we do then no matter if current node matches request path, but does not have a route assigned, other routes will be tested (and we know that path will never match because it does now)

Reply:
But how can you assign a middleware to a node without a route ? What would the middleware be mapped to if we assume that a middleware is a key:value pair of route:path ?